org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/client/PathWarnings.java

```diff
@@ -0,0 +1,58 @@
+package org.eclipse.lemminx.client;
+
+import java.io.File;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.lemminx.customservice.ActionableNotification;
+import org.eclipse.lemminx.customservice.XMLLanguageClientAPI;
+import org.eclipse.lemminx.settings.SharedSettings;
+import org.eclipse.lsp4j.Command;
+import org.eclipse.lsp4j.MessageParams;
+import org.eclipse.lsp4j.MessageType;
+
+public class PathWarnings {
```

Comment:
A lof of code are similar with LimitExceededWarnings. I suggest that you create an abstract class like `AbstractXMLNotification which manages cache and send of the notification and LimitExceededWarnings and PathWarnings  extend it.

Reply:
fixed